### PR TITLE
 [CrowdStrike Falcon] Additional command to get IOA Rules for custom IOA rule triggered detections.

### PR DESCRIPTION
**Pull request by zeekforit**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Added a command to be able to get IOA Rules for custom IOA rule triggered detections. Allowing this command addition can enhance the detection details specifically for custom rules since we can't identify what rule was triggered using the usual Crowdstrike.Detections context.

Honestly, I can't have this tested on our environment but this is the same way how we do this using another SOAR platform. Our team will be really thankful if this feature has been added. I don't care if I don't get any credits for this or if this code was used. we just need a command that can get IOA Rules. Thank you.

## Must have
- [ ] Tests 
- [ ] Documentation

### CODE REVIEW

**Review by zeekforit — COMMENTED** [review]
Fixed errors

**Review by zeekforit — COMMENTED** [review]
Fixed errors

**Review by zeekforit — COMMENTED** [review]
cleanup

**Comment by yucohen** [comment]
Hi @zeekforit Thanks for your contribution! 
Let's set up a short session to test your changes together :) 
Please reach out here or on DFIR in slack in order to set the session

**Comment by zeekforit** [comment]
Hi @yucohen,

Thank you for your response. Unfortunately, I haven't received the invite for DFIR in slack yet. Wanted to ask how we can proceed on testing?

**Comment by yucohen** [comment]
@zeekforit I sent you an email, lets discuss it there :)

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 1]
```diff
@@ -6536,6 +6549,60 @@ def table_headers_transformer(header: str) -> str:
...
+        outputs_prefix='CrowdStrike.IOARules',
+        outputs_key_field='instance_id',
+        outputs=ioarules,
+        readable_output=tableToMarkdown(
+            name='CrowdStrike IOA Rules',
+            t=ioarules,
+            headers=['instance_id', 'customer_id', 'action_label', 'comment', 'committed_on', 'created_by', 'created_on',
+                     'deleted', 'description', 'disposition_id', 'enabled', 'field_values', 'instance_version', 'magic_cookie',
+                     'modified_by', 'modified_on', 'name', 'pattern_id', 'pattern_severity',
+                     'rulegroup_id', 'ruletype_id', 'ruletype_name', 'version_ids'],
```
Comment (yucohen) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:6597]:
```suggestion
        readable_output=tableToMarkdown(
            name='CrowdStrike IOA Rules',
            t=ioarules,
            headers=['instance_id', 'description', 
            'enabled', 'name', 'pattern_id'],
```

Comment (yucohen) [reply Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:6597]: We can narrow down the table, all relevant data would be in the context

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.yml [hunk 2]
```diff
@@ -4884,7 +4884,84 @@ script:
     - contextPath: CrowdStrike.IncidentBehavior.compound_tto
       description: The compound TTO.
       type: String
-  dockerimage: demisto/py3-tools:1.0.0.84136
+  - arguments:
+    - description: The rule ID to get IOA rules for. This can be acquired by combining cid and rule_instance_id from Crowdstrike.Detections context for Custom IOA Rules detections.
+      name: rule_id
```
Comment (yucohen) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.yml:4889]:
```suggestion
      name: rule_ids
```

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 3]
```diff
@@ -6536,6 +6549,60 @@ def table_headers_transformer(header: str) -> str:
...
+            "name": "Name",
+            "pattern_id": "Pattern ID",
+            "pattern_severity": "Pattern Severity",
+            "rulegroup_id": "Rule Group ID",
+            "ruletype_id": "Rule Type ID",
+            "ruletype_name": "Rule Type Name",
+            "version_ids": "Version IDs",
+        }
+
+        return mapping.get(header, header)
```
Comment (yucohen) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:6585]:
```suggestion
  
```

Comment (yucohen) [reply Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:6585]: we can use `headerTransform=string_to_table_header` instead

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 4]
```diff
@@ -6536,6 +6549,60 @@ def table_headers_transformer(header: str) -> str:
...
+        outputs_key_field='instance_id',
+        outputs=ioarules,
+        readable_output=tableToMarkdown(
+            name='CrowdStrike IOA Rules',
+            t=ioarules,
+            headers=['instance_id', 'customer_id', 'action_label', 'comment', 'committed_on', 'created_by', 'created_on',
+                     'deleted', 'description', 'disposition_id', 'enabled', 'field_values', 'instance_version', 'magic_cookie',
+                     'modified_by', 'modified_on', 'name', 'pattern_id', 'pattern_severity',
+                     'rulegroup_id', 'ruletype_id', 'ruletype_name', 'version_ids'],
+            headerTransform=table_headers_transformer,
```
Comment (yucohen) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:6598]:
```suggestion
            headerTransform=string_to_table_header,
```

Comment (yucohen) [reply Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:6598]: it does the same thing :)

File: Packs/CrowdStrikeFalcon/ReleaseNotes/1_12_11.md [hunk 5]
```diff
@@ -4,4 +4,4 @@
 ##### New: CrowdStrike Falcon - Block File
 
 New: This playbook receives an MD5 or a SHA256 hash and adds it to the block list in CrowdStrike Falcon. 
-The playbook uses the integration "CrowdStrike Falcon". (Available from Cortex XSOAR 6.10.0).
+The playbook uses the integration "CrowdStrike Falcon". (Available from Cortex XSOAR 6.10.0).
```
Comment (yucohen) [inline Packs/CrowdStrikeFalcon/ReleaseNotes/1_12_11.md:7]: lets revert this change

File: Packs/CrowdStrikeFalcon/ReleaseNotes/1_12_12.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+
+#### Integrations
+
+##### CrowdStrike Falcon
+- Added 1 new command:
+    - ***cs-falcon-ioarules***
+
+This command allows to retrieve rules for Custom IOA rules detections.
```
Comment (yucohen) [inline Packs/CrowdStrikeFalcon/ReleaseNotes/1_12_12.md:8]:
```suggestion
- Added the ***cs-falcon-ioarules*** command that allows to retrieve rules for Custom IOA rules detections.
```

**Review by yucohen — CHANGES_REQUESTED** [review]
Great work! Please see my comments

**Comment by zeekforit** [comment]
Hello @yucohen

All needed changes were made. Thank you so much for the extended help you've made to achieve this.

**Review by zeekforit — COMMENTED** [review]
All needed changes were made. Thank you so much for the extended help you've made to achieve this.

**Review by zeekforit — COMMENTED** [review]
.

**Review by zeekforit — COMMENTED** [review]
update docker image

**Review by zeekforit — COMMENTED** [review]
.

**Review by yucohen — APPROVED** [review]

**Pull request merged**